src/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -679,6 +682,7 @@
     <Compile Include="System\Runtime\InteropServices\StructLayoutAttribute.cs" />
     <Compile Include="System\WeakReference.cs" />
     <Compile Include="System\WeakReferenceOfT.cs" />
+    <Compile Include="..\..\Common\src\Internal\Text\UTF8String.cs" />
```

Comment:
I do not think you need to pull the UTF8String into CoreLib for this. Instead, LoadLibrary argument on Unix should be regular string instead `byte*` and we should leverage the default string marshalling for it.